### PR TITLE
Allow embedders to update preferrred locales.

### PR DESCRIPTION
**Pull request by chinmaygarde**
This was previously only possible by sending an undocumented payload over an engine managed channel.

### CODE REVIEW

File: shell/platform/embedder/embedder.h [hunk 1]
```diff
@@ -1414,6 +1438,24 @@ FlutterEngineResult FlutterEngineRunTask(FLUTTER_API_SYMBOL(FlutterEngine)
...
                                          const FlutterTask* task);
 
+//------------------------------------------------------------------------------
+/// @brief      Notify a running engine instance that the locale has been
+///             updated. The preferred locale must be the first item in the list
+///             of locales supplied. The other entries will be used as a
+///             fallback.
+///
+/// @param[in]  engine         A running engine instance.
+/// @param[in]  locales        The updates locales in the order of preference.
```
Comment (GaryQian) [inline shell/platform/embedder/embedder.h]: did you mean `updates` -> `update`

File: shell/platform/embedder/tests/embedder_unittests.cc [hunk 2]
```diff
@@ -2751,5 +2751,63 @@ TEST_F(
...
+  locale1.language_code = "";  // invalid
+  locale1.country_code = "US";
+  locale1.script_code = "";
+  locale1.variant_code = nullptr;
+
+  FlutterLocale locale2 = {};
+  locale2.struct_size = sizeof(locale2);
+  locale2.language_code = "zh";
+  locale2.country_code = "CN";
+  locale2.script_code = "";
```
Comment (GaryQian) [inline shell/platform/embedder/tests/embedder_unittests.cc]: Could you include something to explicitly verify script code such as `Hans` or `Hant` are being successfully passed? We have had problems with these in the past, so good to double check

File: shell/platform/embedder/embedder.h [hunk 3]
```diff
@@ -778,6 +778,30 @@ typedef struct {
   FlutterLayersPresentCallback present_layers_callback;
 } FlutterCompositor;
 
+typedef struct {
+  /// This size of this struct. Must be sizeof(FlutterLocale).
+  size_t struct_size;
+  /// The language code of the locale. For example, "en". This is a required
+  /// field. The string must be null terminated. It may be collected after the
+  /// call the `FlutterEngineUpdateLocales`.
```
[the remark this answers is not in the document]
Comment (chinmaygarde) [reply shell/platform/embedder/embedder.h]: Done.

File: shell/platform/embedder/embedder.h [hunk 4]
```diff
@@ -778,6 +778,30 @@ typedef struct {
...
+  const char* language_code;
+  /// The country code of the locale. For example, "US". This is a an optional
+  /// field. The string must be null terminated if present. It may be collected
+  /// after the call the `FlutterEngineUpdateLocales`. If not present, a
+  /// `nullptr` may be specified.
+  const char* country_code;
+  /// The script code of the locale. This is a an optional field. The string
+  /// must be null terminated if present. It may be collected after the call the
```
[the remark this answers is not in the document]
Comment (chinmaygarde) [reply shell/platform/embedder/embedder.h]: Done.

**Pull request merged**